src/app.ts

```diff
@@ -141,38 +159,38 @@ for (const file of monitorFiles) {
 // Load other client events
 require('./events');
 
-client.on('ready', async () => {
-	if (!client || !client.user) throw new Error(`Bot not logged in and ready event triggered.`); // Should never happen
+client.on('ready', () => void (async () => {
+	if (!client?.user) throw new Error(`Bot not logged in and ready event triggered.`); // Should never happen
 	console.log(`Logged in as ${client.user.tag}.`);
 
 	// Startup events
 
 	await Promise.all([
-		...commands.map(cmd => {
-			// Aliases cannot be initalized so we skip them
-			if (typeof cmd === 'function') cmd.init();
-		}),
+		...commands
+			// Aliases cannot be initialized so we skip them
+			.filter(cmd => typeof cmd === 'function')
+			.map(cmd => (cmd as Constructable<BaseCommand>).init()),
 		...monitors.map(monitor => monitor.init()),
 	]);
-});
+})());
 
 // Fires when we get a new message from discord. We ignore messages that aren't commands or are from a bot.
-client.on('message', async msg => {
+client.on('message', (m) => void (async msg => {
```

Comment:
Not understanding this change, hoping to learn something new today at least.

Reply:
`client.on` expects a synchronous callback, not an async one, so I wrapped your async code in a wrapper function. This is pretty dirty, and not how we'd ideally do things.

Reply:
Whats the better way to do this? I presume that would be changing the architecture to be fully synchronous.

Reply:
Not necessarily; I'd either switch to an API that allows async callbacks or pull out everything after `verifyData` to some other function. But yeah, if you want to use a Discord API with synchronous callback you'll either have to `void` a lot of things or use synchronous code.